src/main/scala/edu/berkeley/ce/sparkrocks/Block.scala

```diff
@@ -446,84 +515,54 @@ case class Block(center: Array[Double], faces: Seq[Face], generation: Int=0) ext
   }
 
   /**
-    * Mesh the faces using Delaunay triangulation. This meshing is done
-    * in order to calculate the volume and centroid of the block
+    * Get the counter-clockwise oriented vertices of each face of the block
     *
-    * @return A Seq of seqs that give the local to global mapping for the
-    * triangulation of each of the faces.
+    * @return A mapping from each face of the block to a Seq of vertices for that face
+    *         in a counter-clockwise orientation.
+    *         This function should only be called once all redundant faces have been removed.
     */
-  def meshFaces(vertices: Map[Face, Seq[Array[Double]]]): Map[Face, Seq[Array[Int]]] =
-    // Determine rotation matrix to rotate faces perpendicular to z-axis. This way all vertices
-    // are only in the x-y plane which makes triangulation easier.
-    faces.zip (
-      faces.map { face =>
+  def orientedVertices: Map[Face, Seq[Array[Double]]] = {
+    val faceVertices = calcVertices
+    faceVertices.keys.zip(
```

Comment:
Now that I think about it, you should be able to express this in a more idiomatic way. I believe you can just call `map` (or better yet `transform`) on the `Map` instance you get from `calcVertices` instead of using `keys.zip` and `toMap`.